### PR TITLE
Address feedbacks: Rename type, improve readme

### PR DESCRIPTION
**Pull request by zolvarga**

### CODE REVIEW

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py [hunk 1]
```diff
@@ -599,6 +600,9 @@ def delete_event_route(self, event_route_id, **kwargs):
     def query_twins(self, query_expression, **kwargs):
         # type: (str, **Any) -> ~azure.core.async_paging.ItemPaged[Dict[str, object]]
         """Query for digital twins.
+            Note: that there may be a delay between before changes in your instance are reflected in queries.
```
Comment (anthonyvercolano) [inline sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py:603]: > between [](start = 43, length = 8)

Either something missing here or extra word.

**Comment by anthonyvercolano** [comment]
>         :raises :class: `~azure.core.exceptions.ResourceNotFoundError`: One or more of

not found seems like an odd error for something that already exists.

---
Refers to: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/_digitaltwins_client_async.py:487 in 180e8bd. [](commit_id = 180e8bda49bc2ee9f024928cf0c4672be036eefd, deletion_comment = False)

File: sdk/digitaltwins/azure-digitaltwins-core/samples/Readme.md [hunk 2]
```diff
@@ -139,6 +139,9 @@ print(get_twin)
 
 Query the Azure Digital Twins instance for digital twins using the [Azure Digital Twins Query Store lanaguage](https://review.docs.microsoft.com/azure/digital-twins/concepts-query-language). Query calls support paging. Here's an example of how to query for digital twins and how to iterate over the results.
 
+Note that there may be a delay between before changes in your instance are reflected in queries.
```
Comment (anthonyvercolano) [inline sdk/digitaltwins/azure-digitaltwins-core/samples/Readme.md:142]: >between [](start = 31, length = 7)

either extra word or something missing.

**Review by anthonyvercolano — DISMISSED** [review]
:shipit:

**Comment by anthonyvercolano** [comment]
I approve but I imagine you will wait for the real experts here.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py [hunk 3]
```diff
@@ -563,11 +564,11 @@ def list_event_routes(self, **kwargs):
 
     @distributed_trace
     def upsert_event_route(self, event_route_id, event_route, **kwargs):
-        # type: (str, "models.EventRoute", **Any) -> None
+        # type: (str, "DigitalTwinsEventRoute", **Any) -> None
         """Create or update an event route.
 
         :param str event_route_id: The Id of the event route to create or update.
-        :param ~azure.digitaltwins.models.EventRoute event_route: The event route data.
+        :param DigitalTwinsEventRoute event_route: The event route data.
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py:571]: In docstrings, you should use the fully qualified name to enable hyper-linking:
`:param ~azure.digitaltwins.DigitalTwinsEventRoute:`

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_models.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    :keyword str id: The id of the event route.
+    :keyword str endpoint_name: The name of the endpoint this event route is bound to.
+    :keyword str filter: An expression which describes the events which are routed to the endpoint.
+    """
+
+    def __init__(self, **kwargs):
+        self.id = None
+        self.endpoint_name = kwargs['endpoint_name']
+        self.filter = kwargs['filter']
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_models.py:27]: Are these required parameters? If so, they should be positional parameter in the constructor signature.
If they're optional parameters, then you will need to use `kwargs.get`.

Is the intention here to make the `id` attribute "read only"?

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_models.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    
+    @classmethod
+    def _from_generated(cls, generated):
+        if not generated:
+            return cls()
+        return cls(
+            id=generated.id,
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_models.py:34]: The constructor above doesn't support `id` as a parameter. You would need something like:
```python
obj = cls(endpoint_name=generated.endpoint_name, filter=generated.filter)
obj.id = generated.id
return obj
```

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/__init__.py [hunk 6]
```diff
@@ -7,5 +7,15 @@
 # --------------------------------------------------------------------------
 
 from ._digitaltwins_client_async import DigitalTwinsClient
+from ._models import DigitalTwinsEventRoute
+from .._generated.models import DigitalTwinsModelData
+from .._generated.models import QueryResult
+from .._generated.models import IncomingRelationship
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/__init__.py:13]: These models shouldn't be added to the `aio` namespace.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py [hunk 7]
```diff
@@ -524,12 +525,12 @@ def delete_model(self, model_id, **kwargs):
...
     def get_event_route(self, event_route_id, **kwargs):
-        # type: (str, **Any) -> ~azure.digitaltwins.models.EventRoute
+        # type: (str, **Any) -> DigitalTwinsEventRoute
         """Get an event route.
 
         :param str event_route_id: The Id of the event route.
-        :return: The EventRoute object.
-        :rtype: ~azure.digitaltwins.models.EventRoute
+        :return: The DigitalTwinsEventRoute object.
+        :rtype: DigitalTwinsEventRoute
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py:533]: You will need to actually wrap the result in your new class wrapper:
```python
result = self._client.event_routes.get_by_id(event_route_id, **kwargs)
return DigitalTwinsEventRoute._from_generated(result)
```

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py [hunk 8]
```diff
@@ -541,13 +542,13 @@ def get_event_route(self, event_route_id, **kwargs):
...
-        # type: (**Any) -> ~azure.core.paging.ItemPaged[~azure.digitaltwins.models.EventRoute]
+        # type: (**Any) -> ~azure.core.paging.ItemPaged[DigitalTwinsEventRoute]
         """Retrieves all event routes.
 
         :keyword int results_per_page: The maximum number of items to retrieve per request.
             The server may choose to return less than the requested max.
-        :return: An iterator instance of list of EventRoute.
-        :rtype: ~azure.core.paging.ItemPaged[~azure.digitaltwins.models.EventRoute]
+        :return: An iterator instance of list of DigitalTwinsEventRoute.
+        :rtype: ~azure.core.paging.ItemPaged[DigitalTwinsEventRoute]
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py:551]: You will need to wrap the result here too, I think something like this should work:
```
return self._client.event_routes.list(
    event_routes_list_options=event_routes_list_options,
    cls=lambda objs: [DigitalTwinsEventRoute._from_generated(e) for e in objs]),
    **kwargs
)
```

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/_models.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
+# -------------------------------------------------------------------------
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/_models.py:1]: We don't need this file

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py (hunk 8 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py:565]: Not anymore. I renamed the type with a autorest transformation.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/_models.py (hunk 9 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/_models.py]: Removed

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py (hunk 7 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py:547]: Not anymore. I renamed the type with a autorest transformation.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py (hunk 1 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py:617]: I think it is OK. (It is copied from the C# comment)

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py (hunk 3 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_digitaltwins_client.py:585]: Fixed

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_models.py (hunk 4 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_models.py]: This file has been removed.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_models.py (hunk 5 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/_models.py]: This file has been removed.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/__init__.py (hunk 6 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/__init__.py]: Removed.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/__init__.py [hunk 10]
```diff
@@ -5,7 +5,3 @@
 # Code generated by Microsoft (R) AutoRest Code Generator.
 # Changes may cause incorrect behavior and will be lost if the code is regenerated.
 # --------------------------------------------------------------------------
-
-from ._digitaltwins_client_async import DigitalTwinsClient
-
-__all__ = ['DigitalTwinsClient']
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/__init__.py:11]: This shouldn't have been removed - the client still needs to be exported in the `aio` namespace.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/_digitaltwins_client_async.py [hunk 11]
```diff
@@ -573,11 +587,11 @@ async def list_event_routes(self, **kwargs):
 
     @distributed_trace_async
     async def upsert_event_route(self, event_route_id, event_route, **kwargs):
-        # type: (str, "models.EventRoute", **Any) -> None
+        # type: (str, ~azure.digitaltwins.models.DigitalTwinsEventRoute, **Any) -> None
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/_digitaltwins_client_async.py:590]: This syntax (~azure.digitaltwind.DigitalTwinsEventRoute) should only be used in the docstrings.
For the type hints, you can use the imported model directly.

File: sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,12 @@
 # Release History
 
+## 1.0.0 (2020-10-30)
+
+- The is the GA release containing the following changes:
+  - Expose if_none_match optional parameters for upsert_digital_twin and upsert_relationship APIs
+  - Rename EventRoute type to DigitalTwinsEventRoute
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md:7]: We've also renamed parameters `component_name` and `dtdl_models`

File: sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,12 @@
 # Release History
 
+## 1.0.0 (2020-10-30)
+
+- The is the GA release containing the following changes:
+  - Expose if_none_match optional parameters for upsert_digital_twin and upsert_relationship APIs
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md:6]: We could probably rephrase this to something like:
- Added `etag` and `match_condition` parameters to `upsert_digital_twin` and `upsert_relationship` APIs to support conditional operation.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/__init__.py (hunk 10 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/__init__.py:11]: Fiexed.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/_digitaltwins_client_async.py (hunk 11 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/aio/_digitaltwins_client_async.py:590]: Fixed.

File: sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md (hunk 13 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md:6]: Thanks - Added.

File: sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md (hunk 12 above)
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md:7]: Added.

File: sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/__init__.py [hunk 14]
```diff
@@ -10,14 +10,13 @@
 from ._generated.models import DigitalTwinsModelData
 from ._generated.models import QueryResult
 from ._generated.models import IncomingRelationship
-from ._generated.models import EventRoute
```
[the remark this answers is not in the document]
Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/azure/digitaltwins/core/__init__.py:13]: Fixed.

File: sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,14 @@
 # Release History
 
+## 1.0.0 (2020-10-30)
```
Comment (annatisch) [inline sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md:3]: Can you leave the date as `(unreleased)` for now?
We'll populate it when needed.

Comment (zolvarga) [reply sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md]: Fixed.

**Review by annatisch — APPROVED** [review]
